mmdet/models/backbones/resnet.py

```diff
@@ -235,19 +275,17 @@ def __init__(self,
         assert max(out_indices) < num_stages
         self.style = style
         self.frozen_stages = frozen_stages
-        self.bn_eval = bn_eval
-        self.bn_frozen = bn_frozen
+        assert (isinstance(normalize, dict) and 'eval_mode' in normalize
+                and 'frozen' in normalize)
+        self.norm_eval = normalize.pop('eval_mode')
```

Comment:
We can add a default value for `eval_mode` here.

Reply:
Yes I know that. I mean we can just left `eval_mode(True)` and `frozen(False)` optional. Besides, the default value of L34 is `normalize=dict(type='BN')`.

Reply:
I intentionally add `normalize=dict(type, eval_mode, frozen)` to let users know how to specify `normalize` dict in config. If we leave `eval_mode(True)` and `frozen(False)` in the body, it will be subtle to add norm mode in to config files.

Reply:
I think it is perfect to use `normalize=dict(type, eval_mode, frozen)` as the default value, which can be a good indicator. It may be better to also allow users to modify configs like `normalize=dict(type='BN', eval_mode=False)` instead of writing out all parameters.

Reply:
`eval_mode` and `frozen` is mandatory for the backbone. But for other modules (neck, head), just leave `normalize=dict(type='BN')` is oke because `frozen` is optional in `build_norm_layer`. Btw, it is noted that `eval_mode` in neck or head is not supported since it will be overridden by parent modules and there will be error informed.

Reply:
Okay we can leave the assertion as it is. For necks and heads, there is no pretrained weights so `eval_mode` is usually set to False. Maybe leave a comment for potential errors?

Reply:
My idea is add following into `build_norm_layer`

```
if eval_mode in normalize:
    raise Exception('eval_mode is supported in backbone only and prohibited in other modules')
```

Do you have any suggestion?

Reply:
I think we don't need a strict error here, maybe just a comment. Besides backbones, there are indeed some cases that other modules also have pretrained weights, e.g., res5 head.

Reply:
Oh. i didnt notice that. However, I think it should be an error because if other modules have pretrained weights, `eval_mode` should be popped out and processed in the `train()` function, not inside `build_norm_layer`. I think the error message here should be changed to `eval_mode for modules without pretrained weights is not supported` to avoid accidentally adding `eval_mode` to non-pretrained modules.

Reply:
Looks good to me.

Reply:
After writing some config files, I find that putting `eval_mode` in `normalize` is not a good idea. It may be better to follow the original style to use a variable `norm_eval` in the modules where normalization layers may need to be set as eval mode, e.g., backbones.

In this way, we can use a shared variable in the config file.
```python
normalize = xxxx
models = dict(
backbones=(normalize=normalize, xxx),
neck=(normalize=normalize, xxx),
bbox_head=(normalize=normalize, xxx),
mask_head=(normalize=normalize, xxx)
)
```
Also, we do not need to check the `eval_mode` field in `build_norm_layer`.

Reply:
It is a nice point I think.